toolkit/Makefile

```diff
@@ -19,16 +21,29 @@ PACKAGE_REBUILD_LIST    ?=
 PACKAGE_IGNORE_LIST     ?=
 SRPM_PACK_LIST          ?=
 
-REBUILD_TOOLCHAIN               ?= n
+# Logic to auto configure build options for optimized builds
+# These will set the default values for: REBUILD_TOOLS, REBUILD_TOOLCHAIN, DELTA_BUILD, INCREMENTAL_TOOLCHAIN, and ALLOW_TOOLCHAIN_DOWNLOAD_FAIL
+include ./quickrebuild_autoconfig.mk
+
+# We generally want to use the DELTA_BUILD flag to determine whether to use the incremental toolchain, but to preserve
+# backwards compatibility, we'll still use the INCREMENTAL_TOOLCHAIN flag if it's set.
+DELTA_BUILD                     ?= n
+ifeq ($(origin DELTA_BUILD), undefined)
+# We didn't set DELTA_BUILD, so use the previous default behavior of INCREMENTAL_TOOLCHAIN for backwards compatibility.
 INCREMENTAL_TOOLCHAIN           ?= n
+else
+# We set DELTA_BUILD, so inherit value for INCREMENTAL_TOOLCHAIN.
+INCREMENTAL_TOOLCHAIN           ?= $(DELTA_BUILD)
+endif
+
+REBUILD_TOOLCHAIN               ?= n
 ALLOW_TOOLCHAIN_DOWNLOAD_FAIL   ?= n
 UPDATE_TOOLCHAIN_LIST           ?= n
 REBUILD_PACKAGES                ?= y
 DOWNLOAD_SRPMS                  ?= n
 ALLOW_SRPM_DOWNLOAD_FAIL        ?= n
 REBUILD_TOOLS                   ?= n
```

Comment:
With the new mk file the `REBUILD_TOOLS` and a few other arguments now have two places where their defaults are set. I think this is making it even harder to understand what's going on. Could we keep the defaults in one place? In this case move these `?=` to the end of the new mk file, so that we have some values set for them, if they none of the quick build branches touched them?

Reply:
Sure, I'll move everything related into the new file.

Reply:
Maybe call it `incremental_builds.mk`.